typescript.md

```diff
@@ -204,10 +223,60 @@ export interface User { ... }
 ```ts
 interface Building {
   room: {
-    door: string,
-    walls: string[],
+    door: string;
+    walls: string[];
   };
 }
 
 type Walls = Building['room']['walls']; // string[]
 ```
+
+## Keyof Type Operator
+
+```ts
+type Point = { x: number; y: number };
+
+type P = keyof Point; // x | y
+```
+
+## Conditinal Types
+
+```ts
+// SomeType extends OtherType ? TrueType : FalseType;
+
+type ToArray<T> = T extends any ? T[] : never;
+
+type StrArrOrNumArr = ToArray<string | number>; // string[] | number[]
+```
+
+### Inferring
+
+```ts
+type GetReturnType<T> = T extends (...args: unknown[]) => infer R
+  ? R
+  : never;
+
+type Num = GetReturnType<() => number>; // number
+
+type First<T extends Array<any>> = T extends [infer F, ...infer Rest] ? F : never;
+
+type Str = First<['hello', 1, false]>; // 'hello'
+```
```

Comment:
those are two different examples. You may split those.

```ts
type GetReturnType<T> = T extends (...args: unknown[]) => infer R
  ? R
  : never;

type Num = GetReturnType<() => number>; // number
```

```ts
type First<T extends Array<any>> = T extends [infer F, ...infer Rest] ? F : never;

type Str = First<['hello', 1, false]>; // 'hello'
```

Reply:
Thanks.
I fixed it.